pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnMultiValueReaderWriter.java

```diff
@@ -68,8 +94,10 @@
 
   private PinotDataBuffer headerBuffer;
   private List<PinotDataBuffer> dataBuffers = new ArrayList<>();
-  private FixedByteSingleValueMultiColWriter headerWriter;
-  private FixedByteSingleValueMultiColReader headerReader;
+  private List<PinotDataBuffer> headerBuffers = new ArrayList<>();
```

Comment:
Since this is a list of buffers, what's the order of magnitude of the typical length of that list? Is it low (eg. 1-10 entries) or something large like 100+ entries?

You'll want to keep in mind that if this is large, you'll want to coalesce buffers as I believe that large allocations are implemented as anonymous `mmap`s; as such, the process could exhaust `vm.max_map_count` and cause the JVM to crash.